src/SearchNames.js

```diff
@@ -0,0 +1,38 @@
+import React, { useState} from "react";
+import SortByGender from "./SortByGender";
+
+const handleChange = (setText, text, sortedNames, setFilteredNames, event) => {
+    setText(event.target.value);
+    filterNames(text, sortedNames, setFilteredNames);
+    event.preventDefault();
+};
+
+const filterNames = (text, sortedNames, setFilteredNames) => {
+    const searchInput = text.toLowerCase();
+    const searchResult = sortedNames.filter((el) =>
+        el.name.toLowerCase().includes(searchInput)
+    );
+    setFilteredNames(searchResult);
+};
+
+const SearchNames = ( { sortedNames, setFilteredNames } ) => {
+    const [text, setText] = useState("");
+
+    return (
+        <div className="search">
+            <input
+            type="text"
+            value={text}
+            id="search"
+            onChange={handleChange.bind(this, setText, text, sortedNames, setFilteredNames)}
```

Comment:
Tbh, I have never used `bind()` and `this` functionality before, so I'm not entirely sure why it was used in this context. It would be helpful if you could explain the reasoning behind its usage.

Just FYI, you can simplify the definition of your handleChange function. All you need to do is update the `setText` state with the value obtained from `event.target.value` and then call filterNames accordingly.